jdaviz/configs/default/plugins/subset_tools/tests/test_subset_tools.py

```diff
@@ -287,34 +287,31 @@ def test_get_regions_composite(cubeviz_helper, spectrum1d_cube):
     ``app.get_subsets``. This test ensures that a region is returned through
     both ``app.get_subsets`` and ``get_regions``.
     """
-    cubeviz_helper.load_data(spectrum1d_cube)
-    plg = cubeviz_helper.plugins['Subset Tools']
-
-    # For some reason, Subset 2 disappears after the third subset is applied
-    # when loaded this way. Uncomment to replace _apply_interactive_region once
-    # JDAT-5014 is resolved
-    # plg.import_region(CirclePixelRegion(center=PixCoord(x=96.0, y=96.0),
-    #                                     radius=45.0), combination_mode='new')
-    # plg.import_region(CirclePixelRegion(center=PixCoord(x=95.0, y=95.0),
-    #                                     radius=25.0), combination_mode='new')
+    a = np.ones((200, 200))
+    imviz_helper.load_data(a, data_label="test")
+    plg = imviz_helper.plugins['Subset Tools']
 
     # apply two circular subsets
-    cubeviz_helper._apply_interactive_region('bqplot:truecircle', (51, 51), (141, 141))
-    cubeviz_helper._apply_interactive_region('bqplot:truecircle', (70, 70), (120, 120))
+    plg.import_region(CirclePixelRegion(center=PixCoord(x=96.0, y=96.0),
+                                        radius=45.0), combination_mode='new')
+    plg.import_region(CirclePixelRegion(center=PixCoord(x=95.0, y=95.0),
+                                        radius=25.0), combination_mode='new')
 
     # apply composite subset created from two existing circular subsets
-    subset_groups = cubeviz_helper.app.data_collection.subset_groups
+    imviz_helper.app.session.edit_subset_mode._mode = NewMode
```

Comment:
> For some reason, Subset 2 disappears after the third subset is applied

Because the edit mode was not changed to `NewMode`. `_apply_interactive_region` does it for you so that's why it worked by not `import_region`.

Reply:
does combination_mode='new' in import region not take care of this? I expected it to

Reply:
That is for the region it is importing. This is for the next few internal calls that try to combine the first 2 subsets.